validmind/tests/model_validation/statsmodels/GINITable.py

```diff
@@ -65,72 +65,50 @@ class GINITable(Metric):
     }
 
     def run(self):
-        model = (
-            self.inputs.model[0]
-            if isinstance(self.inputs.model, list)
-            else self.inputs.model
-        )
-
-        X_train = self.datasets[0].x
-        y_train = self.datasets[0].y
 
-        X_test = self.datasets[1].x
-        y_test = self.datasets[1].y
-
-        summary_metrics = self.compute_metrics(model, X_train, y_train, X_test, y_test)
+        summary_metrics = self.compute_metrics()
 
         return self.cache_results(
             {
                 "metrics_summary": summary_metrics.to_dict(orient="records"),
             }
         )
 
-    def compute_metrics(self, model, X_train, y_train, X_test, y_test):
-        """Computes AUC, GINI, and KS for train and test sets."""
+    def compute_metrics(self):
+        """Computes AUC, GINI, and KS for an arbitrary number of datasets."""
+        # Initialize the dictionary to store results
+        metrics_dict = {"Dataset": [], "AUC": [], "GINI": [], "KS": []}
 
-        metrics_dict = {"Dataset": ["Train", "Test"], "AUC": [], "GINI": [], "KS": []}
+        # Iterate over each dataset in the inputs
+        for i, dataset in enumerate(self.inputs.datasets):
+            dataset_label = (
+                dataset.input_id
+            )  # Use input_id as the label for each dataset
+            metrics_dict["Dataset"].append(dataset_label)
 
-        for dataset, X, y in zip(
-            ["Train", "Test"], [X_train, X_test], [y_train, y_test]
-        ):
-            y_scores = model.predict(X)
+            # Retrieve y_true and y_pred for the current dataset
+            y_true = np.ravel(dataset.y)  # Flatten y_true to make it one-dimensional
+            y_prob = dataset.y_prob(self.inputs.model.input_id)
 
-            print("Predicted scores obtained...")
```

Comment:
One thing I'd recommend is to use `logger.info` and `logger.debug` versus `print` to print stuff out anywhere in library code. This gives us the opportunity to have different levels of verbosity. So if we are having issues we can optionally set `LOG_LEVEL=debug` to get more verbose output.

It also lets us control logging from a central config which can come in handy :)